src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionManagerImpl.java

```diff
@@ -19,50 +11,79 @@
-        for (ImageTask processedTask: processedTasks) {
-            processedTask.setState(ImageTaskState.REMOTELY_ARCHIVED);
+    public List<Task> addTasks(SubmissionParameters submissionParameters) {
+        List<Date> processedDates = new ArrayList<>();
+        try {
+            List<ImageTask> processedTasks = getAllRemotelyProcessedTasks(submissionParameters);
+            for (ImageTask processedTask : processedTasks) {
+                processedTask.setState(ImageTaskState.REMOTELY_ARCHIVED);
+            }
+            submissionDispatcher.addImageTasks(processedTasks);
+            processedDates = processedTasks.stream()
+                    .map(ImageTask::getImageDate)
+                    .collect(Collectors.toList());
+        } catch (Throwable t) {
+            LOGGER.error("Error while adding remotely processed tasks.", t);
         }
-        submissionDispatcher.addImageTasks(processedTasks);
-        List<Date> datesToExclude = processedTasks.stream()
-                .map(ImageTask::getImageDate)
+        return submissionDispatcher.addTasks(submissionParameters, processedDates);
+    }
+
+    /**
+     * Gets list of processed tasks from all SAPS neighbors from this instance.
+     *
+     * @param submissionParameters Parameters of user submission.
+     * @return List of processed tasks from all SAPS neighbors.
+     */
+    private List<ImageTask> getAllRemotelyProcessedTasks(SubmissionParameters submissionParameters) {
+        String[] SAPSNeighborsUrls = getSAPSNeighborsUrls();
+        List<ImageTask> processedTasks = Arrays.stream(SAPSNeighborsUrls)
+                .map(SAPSNeighborsUrl -> getRemotelyProcessedTasksFromInstance(
+                        SAPSNeighborsUrl,
+                        submissionParameters))
+                .flatMap(Collection::stream)
                 .collect(Collectors.toList());
-        return submissionDispatcher.addTasks(submissionParameters, datesToExclude);
+        return processedTasks;
     }
 
     /**
-     * Gets a list of processed tasks from a SAPS remote instance.
+     * Gets list of processed tasks from SAPS neighbor that had its URL specified.
      *
+     * @param SAPSNeighborUrl SAPS neighbor URL.
      * @param submissionParameters Parameters of user submission.
-     * @return List of processed tasks.
+     * @return List of processed tasks from SAPS neighbor.
      */
-    private List<ImageTask> getRemotelyProcessedTasks(SubmissionParameters submissionParameters) {
+    private List<ImageTask> getRemotelyProcessedTasksFromInstance(String SAPSNeighborUrl,
+                                                                  SubmissionParameters submissionParameters) {
         List<ImageTask> processedTasks = new ArrayList<>();
         try {
-            String remoteInstanceUrl = getRemoteInstanceUrl();
-            ClientResource clientResource = new ClientResource(remoteInstanceUrl);
+            String processedTasksURN = "/archivedTasks";
```

Comment:
move this to class declaration? (also, it seems it deserver private static final

Reply:
Agree, I'll move it